### PR TITLE
Output dots

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Issue: https://github.com/TypedDevs/bashunit/issues/59: Display dots instead of tests names

Currently, the only output from bashunit is the test names in a `prettify way`. 
We want to allow rendering dots (a dot per test). 

## 🔖 Changes

- Render dots (per test) by default
- Render the scripts and test names using the new flag `--verbose`

Screenshot 2023-09-30 at 13 58 46

### CODE REVIEW

**Comment by cmayo** [comment]
I like the output with the dots.

Why don't print the dots as default and show the current output with a verbose option? Maybe it has more sense to show the dots as a short info by default and if someone want more information then set the `--verbose` option. What do you think?

**Comment by Chemaclass** [comment]
> Why don't print the dots as default and show the current output with a verbose option? 

Great idea, @cmayo ! Done ✔️  CC: @khru @Tito-Kati

File: bashunit [hunk 1]
```diff
@@ -34,6 +35,11 @@ while [[ $# -gt 0 ]]; do
       shift
       shift
       ;;
+    --verbose)
```
Comment (cmayo) [inline bashunit:38]: Just to allow also the short option name

```suggestion
    --verbose|-v)
```

Comment (Chemaclass) [reply bashunit:38]: Oh, right! I thought about it, but I forgot it. Thanks!

**Comment by Chemaclass** [comment]
We could think about doing in a follow up PR that if the test fail it will render an `f` instead of a `.` This will be a bit more complex than simply adding this:
```bash
# similar as we just did in this PR with `Console::printSuccessfulTest()`
function Console::printFailedTest() {
  ((_RENDERED_TEST_COUNT++))

  if [[ "$_VERBOSE" == false ]]; then
    if (( _RENDERED_TEST_COUNT % 50 != 0 )); then
      printf "f"
    else
      echo "f"
    fi
  else
    printf "..."
  fi
}
```

Because we should display at the end the actual test(s) that failed, and at this moment the only place where we display the error is inside `Console::printFailedTest()`

**Review by Tito-Kati — APPROVED** [review]

**Review by cmayo — APPROVED** [review]

**Comment by Chemaclass** [comment]
> I like the output with the dots.
> 
> Why don't print the dots as default and show the current output with a verbose option? Maybe it has more sense to show the dots as a short info by default and if someone want more information then set the `--verbose` option. What do you think?

After some rethinking with @khru, we prefer keeping the "pretty" output by default and using the "dots" as optional. The reason is based on the value that you have on both options; when writing tests, the nicer version gives you already more value than the dots. I will create another PR to change this behaviour.

**Pull request merged**